### ISSUE TITLE
Coroutines Hiding Errors Within on_ready Event

### ISSUE THREAD

**Issue by Ashthetik** [brainboxdotcc/DPP#1222]
### Git Commit Reference
ced36fd7652ed24ec3784cbc150722a1e5f6456a

### Describe the bug:
When utilising coroutines within the `on_ready` function, it's possible for D++ to ignore HTTP error codes and skip on printing any errors.
This specifically happens when a user creates a malformed Slash Command (see below) and proceeds to send it to Discord.

### To Reproduce:
Utilising the basic layout of the example bot provided in the [Using Slash Commands and Interactions](https://dpp.dev/slashcommands.html) wiki: 
1. Convert the `on_ready` to a coroutine utilisng `-> dpp::task<void>`.
2. Malform the command data either by leaving a field empty, or overpopulating (>100 characters) -- see Additional Context.
3. Resume building the bot as usual.
4. Observe as it fails to output the error.

### Expected Behaviour:
As expected with a synchronous command creation, it would also be expected that any error codes would also mirror through with a coroutine setup.

### Screenshots:
![DPP_Bug_Report_Ev1](https://github.com/user-attachments/assets/a3da3633-094d-4850-8901-463e227d8db1)

### System Details:
- OS: Linux fedora 6.10.3-200.fc40.x86_64 
- Version: 40

### Additional Context:
Example Minimal Code:
```cpp
#include <dpp/dpp.h>

int main() {
    dpp::cluster bot("...");

    bot.on_ready([&bot, logs](const dpp::ready_t &) -> dpp::task<void> {
        if (dpp::run_once<struct bulk_register>()) {
            const dpp::slashcommand bancmd(
                "", // Empty Field
                // Overpopulated Field
               "...............................................................................................................................................................................................................................................   ", 
                bot.me.id
            );

            co_await bot.co_global_command_create(bancmd);
        }

        co_return;
    });

    bot.start(dpp::st_wait);

    return 0;
}   
```

**Comment by Mishura4** [comment]
Not a bug - you need to print the error yourself with coroutines
The non-coroutine version passes a default callback that prints errors, this wouldn't be feasible with coroutines

With that said we could document this better

**Comment by Ashthetik** [comment]
Noted.
Treating it respectively with `confirmation_callback_t` and `is_error()` mimics it enough.

Never knew coroutines would break the logic flow I had. The more you know :+1:

**Comment by Ashthetik** [comment]
Is it all good to close this, or should I keep it open for tracking purposes for a docs update? I'd love to PR it, but I suck at formal documentation, lol

**Comment by Jaskowicz1** [comment]
Any news on this?

**Comment by braindigitalis** [comment]
is there actually anything to do here? not sure anything needs to be added?

**Comment by Mishura4** [comment]
Documentation

**Comment by Ashthetik** [comment]
Yeah I was supposed to make a PR for the docs on this but forgot

**Comment by Ashthetik** [comment]
All good to close this now with the doc update being merged? Or are we keeping it up for visibility?

**Comment by Jaskowicz1** [comment]
Closed as of #1288
